modules/wolfgame.py

```diff
@@ -787,6 +795,189 @@ def fpinger(cli, nick, chan, rest):
+    else:
+        if acc and acc != "*":
+            var.PING_IF_PREFS_ACCS[acc] = value
+            var.set_ping_if_status_acc(acc, value)
+            if value not in var.PING_IF_NUMS_ACCS.keys():
+                var.PING_IF_NUMS_ACCS[value] = []
+            var.PING_IF_NUMS_ACCS[value].append(acc)
+            if old is not None:
+                if old in var.PING_IF_NUMS_ACCS.keys():
+                    if acc in var.PING_IF_NUMS_ACCS[old]:
+                        var.PING_IF_NUMS_ACCS[old].remove(acc)
+        elif not var.ACCOUNTS_ONLY:
+            var.PING_IF_PREFS[cloak] = value
+            var.set_ping_if_status(cloak, value)
+            if value not in var.PING_IF_NUMS.keys():
+                var.PING_IF_NUMS[value] = []
+            var.PING_IF_NUMS[value].append(cloak)
+            if old is not None:
+                if old in var.PING_IF_NUMS.keys():
+                    if cloak in var.PING_IF_NUMS[old]:
+                        var.PING_IF_NUMS[old].remove(cloak)
+
+@cmd("pingpref", "pingprefs", "prefping", pm=True) # need a better name
+def toggle_altping_prefs(cli, nick, chan, rest):
+    if nick in var.USERS:
+        cloak = var.USERS[nick]["cloak"]
+        acc = var.USERS[nick]["account"]
+    else: # something not right
+        if chan == nick:
+            pm(cli, nick, "You need to be in {0} to use that command.".format(botconfig.CHANNEL))
+        else: # former message: "You won the lottery! This is a bug though, so report it to the admins."
+            cli.notice(nick, "You need to be in {0} to use that command.".format(botconfig.CHANNEL))
+    if acc and acc != "*":
+        if acc in var.PING_PREFS_ACCS.keys() and var.PING_PREFS_ACCS[acc] == "all":
+            msg = "You will now only get pinged by {0}ping when your preferred amount of players is reached."
+            var.PING_PREFS_ACCS[acc] = "none"
+            var.set_ping_pref_acc(acc, "none")
+        else:
+            msg = "You will now get pinged when your preferred amount of players is reached."
+            var.PING_PREFS_ACCS[acc] = "all"
+            var.set_ping_pref_acc(acc, "all")
+    elif var.ACCOUNTS_ONLY:
+        msg = "You are not logged in to NickServ."
+    else:
+        if cloak in var.PING_PREFS.keys() and var.PING_PREFS[cloak] == "all":
+            msg = "You will now only get pinged by {0}ping when your preferred amount of players is reached."
+            var.PING_PREFS[cloak] = "none"
+            var.set_ping_pref(cloak, "none")
+        else:
+            msg = "You will now get pinged when your preferred amount of players is reached."
+            var.PING_PREFS[cloak] = "all"
+            var.set_ping_pref(cloak, "all")
+
+    if chan == nick:
+        pm(cli, nick, msg.format(botconfig.CMD_CHAR))
+    else:
+        cli.notice(nick, msg.format(botconfig.CMD_CHAR))
+
+def join_timer_handler(cli):
+    while var.PHASE == "join":
+        with var.GRAVEYARD_LOCK:
```

Comment:
Well, it doesn't matter at all, since both threads that use this are mutually exclusive; one runs when a game is not running, the other runs when a game is running. It seemed pointless to create a new lock that is only used for that, rather than using one which is already there and unused at the time we need to use it.